### PR TITLE
Move charts to sub-folder

### PR DESCRIPTION
**Pull request by pat-s**
- Rename subcharts: prefix with `woodpecker-`
- Adapt release config
- Add top-level README

Given that the next release is major anyway, the renaming of the sub-charts would fit well now.

### CODE REVIEW

**Comment by pat-s** [comment]
@anbraten 

This now works with a single call to `helm-docs`. I've added a sub-chart template for `woodpecker-server` to include the upgrading notes. `woodpecker-agent` has none and processes the default implicitly

**Comment by anbraten** [comment]
> Rename subcharts: prefix with woodpecker-

Why do you want to rename those? It would in addition collide with the old separate charts again.

**Comment by pat-s** [comment]
> Why do you want to rename those? It would in addition collide with the old separate charts again.

Mainly to have unique names as `agent` and `server` are generic ones. And because 1.0 is a major upgrade anyhow. 

>  It would in addition collide with the old separate charts again.

I wonder if we should clean them up and remove them - as most likely almost nobody ever used them or has meanwhile upgraded?

**Comment by pat-s** [comment]
I reverted the renaming.

File: charts/woodpecker/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Woodpecker Helm Charts
+
+This repository contains the helm charts of the [Woodpecker](https://woodpecker-ci.org) project.
```
Comment (anbraten) [inline charts/woodpecker/README.md:3]: Not sure if this is the expected content.

File: charts/woodpecker/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+![Version: 0.4.2](https://img.shields.io/badge/Version-0.4.2-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 1.0.2](https://img.shields.io/badge/AppVersion-1.0.2-informational?style=flat-square)
+
+**Source Code**: <https://github.com/woodpecker-ci/woodpecker>
+
+## Installation
```
Comment (anbraten) [inline charts/woodpecker/README.md:9]: Should be part of the top level readme I guess

File: charts/woodpecker/README.md.gotmpl [hunk 3]
```diff
@@ -61,20 +45,7 @@ resource "helm_release" "woodpecker" {
...
 <summary>To 1.0.0</summary>
 
-* split `metrics.enabled` and the PodMonitor deployment to `prometheus.podmonitor.enabled`
-* move `metrics.interval` to `prometheus.podmonitor.interval`
+- split `metrics.enabled` and the PodMonitor deployment to `prometheus.podmonitor.enabled`
+- move `metrics.interval` to `prometheus.podmonitor.interval`
 
 </details>
-
-## Versioning
```
Comment (anbraten) [inline charts/woodpecker/README.md.gotmpl:69]: Shouldn't this part be moved to the top level Readme?

File: charts/woodpecker/charts/agent/README.md [hunk 4]
```diff
@@ -32,14 +32,14 @@ A Helm chart for the Woodpecker agent
...
-| env.WOODPECKER_SERVER | string | `"woodpecker-server:9000"` | Add the environment variables for the agent component |
+| env.WOODPECKER_SERVER | string | `"server:9000"` | Add the environment variables for the agent component |
 | extraSecretNamesForEnvFrom | list | `["woodpecker-secret"]` | Add extra secret that is contains environment variables |
 | extraVolumeMounts | list | `[]` | Additional volumes that will be attached to the agent container |
 | extraVolumes | list | `[]` | Additional volumes that can be mounted in containers |
 | fullnameOverride | string | `""` | Overrides the full name of the chart of the agent component |
 | image.pullPolicy | string | `"IfNotPresent"` | The pull policy for the image |
 | image.registry | string | `"docker.io"` | The image registry |
-| image.repository | string | `"woodpeckerci/woodpecker-agent"` | The image repository |
+| image.repository | string | `"woodpeckerci/agent"` | The image repository |
```
Comment (anbraten) [inline charts/woodpecker/charts/agent/README.md:42]: wrong repo name

**Review by anbraten — COMMENTED** [review]
looked through most of the changes. most parts of the readmes you removed should go back to the top level readme IMO. The renaming of the values from woodpecker-server to server is a breaking change (still not sure if necessary). At least please check the replacements again as there are some places were it is wrong as for the image names

File: charts/woodpecker/README.md (hunk 2 above)
Comment (pat-s) [reply charts/woodpecker/README.md:9]: I kept the top-level README tidy and only added links to the chart README. That's also how others like bitnami do it, i.e. have the chart specific READMEs at the root of the chart. And in this case, the installation instructions would go here.

File: charts/woodpecker/README.md (hunk 1 above)
Comment (pat-s) [reply charts/woodpecker/README.md:3]: Agreed, removed it. It's present in the top-level README anyhow.

File: charts/woodpecker/README.md.gotmpl (hunk 3 above)
Comment (pat-s) [reply charts/woodpecker/README.md.gotmpl:69]: I don't think so, see https://github.com/woodpecker-ci/helm/pull/101/files#r1392551144

**Comment by pat-s** [comment]
@anbraten made a bunch of fixes/updates, see recent commits.

File: charts/woodpecker/README.md (hunk 2 above)
Comment (anbraten) [reply charts/woodpecker/README.md:7]: Okay for now. Maybe we can add an extra link to the top-level to direct users to this chart as I assume users will read the top-level first and wont know there is more info on the woodpecker chart.

Comment (pat-s) [reply charts/woodpecker/README.md:7]: Links to the chart READMEs are already present in the top-level README. Or do you mean something else and we misunderstand each other?

File: charts/woodpecker/charts/agent/values.yaml [hunk 5]
```diff
@@ -131,4 +131,4 @@ topologySpreadConstraints: []
 #   whenUnsatisfiable: "DoNotSchedule"
 #   labelSelector:
 #     matchLabels:
-#       "app.kubernetes.io/name": woodpecker-agent
+#       "app.kubernetes.io/name": agent
```
[the remark this answers is not in the document]
Comment (pat-s) [reply charts/woodpecker/charts/agent/values.yaml:134]: Leftover. Well spotted 👍

**Comment by pat-s** [comment]
@anbraten approved?

File: charts/woodpecker/README.md (hunk 2 above)
Comment (anbraten) [reply charts/woodpecker/README.md:7]: I was mainly referring to new users and how they would know where to click to get started. That's why I would personally leave some more context in the top-level Readme, but we can still improve that later again in a new PR

**Review by anbraten — APPROVED** [review]

**Pull request merged**